packages/lib-classifier/src/components/Classifier/Classifier.js

```diff
@@ -44,13 +44,20 @@ function Classifier({
     }
   }
 
+  /** Runs when locale passed from app-project changes */
   useEffect(function onLocaleChange() {
     if (locale) {
       classifierStore.setLocale(locale)
       i18n.changeLanguage(locale)
     }
   }, [locale])
```

Comment:
My hunch is that this runs when locale changes (via app-project), but then the `workflowStrings` logic in ClassifierContainer either doesn't fully update the mobx store, or the classifier UI does not re-render to display the new locale workflow strings. Maybe this could be moved to ClassifierContainer?

Reply:
This runs before workflow strings update, because it doesn’t have to wait for an API request to complete. There are some comments in the tests stepping through the props changes when the locale changes. 

Locale changes, triggering this effect and also triggering a request for new translations. A second effect should trigger when those translations are received.

Reply:
When the workflow snapshot refreshes, it updates `classifierStore.workflows.active`. However, the classifier uses `classifierStore.workflowSteps` to show the current task. Maybe the bug behind #5017 is in the code that keeps those two stores in sync? They should be synced by [`autorun`](https://mobx.js.org/reactions.html#autorun).

Reply:
>Maybe this could be moved to ClassifierContainer?

I'll try that in #5038.

Reply:
Locale changes are now handled in the `ClassifierContainer`, in #5038.